server/src/test/java/org/elasticsearch/action/admin/indices/resolve/ResolveIndexTests.java

```diff
@@ -210,20 +223,98 @@ public void testResolveHiddenProperlyWithDateMath() {
+            List<ResolvedDataStream> dataStreams = new ArrayList<>();
+            TransportAction.resolveIndices(
+                new String[] { ".*" },
+                IndicesOptions.LENIENT_EXPAND_OPEN,
+                clusterState,
+                resolver,
+                indices,
+                aliases,
+                dataStreams
+            );
+            // non net-new system indices are allowed even when no system indices are allowed
+            assertThat(indices.stream().map(ResolvedIndex::getName).collect(Collectors.toList()), hasItem(is(".test-system-1")));
+        }
+        {
+            threadContext = createThreadContext();
+            threadContext.putHeader(EXTERNAL_SYSTEM_INDEX_ACCESS_CONTROL_HEADER_KEY, "test-net-new-system");
+            resolver = new IndexNameExpressionResolver(threadContext, systemIndices);
+            List<ResolvedIndex> indices = new ArrayList<>();
+            List<ResolvedAlias> aliases = new ArrayList<>();
+            List<ResolvedDataStream> dataStreams = new ArrayList<>();
+            TransportAction.resolveIndices(
+                new String[] { ".*" },
+                IndicesOptions.STRICT_EXPAND_OPEN,
+                clusterState,
+                resolver,
+                indices,
+                aliases,
+                dataStreams
+            );
+            assertThat(indices.stream().map(ResolvedIndex::getName).collect(Collectors.toList()), hasItem(is(".test-system-1")));
+            assertThat(indices.stream().map(ResolvedIndex::getName).collect(Collectors.toList()), hasItem(is(".test-net-new-system-1")));
+            indices = new ArrayList<>();
+            TransportAction.resolveIndices(
+                new String[] { ".test-net*" },
+                IndicesOptions.STRICT_EXPAND_OPEN,
+                clusterState,
+                resolver,
+                indices,
+                aliases,
+                dataStreams
+            );
+            assertThat(indices.stream().map(ResolvedIndex::getName).collect(Collectors.toList()), not(hasItem(is(".test-system-1"))));
+            assertThat(indices.stream().map(ResolvedIndex::getName).collect(Collectors.toList()), hasItem(is(".test-net-new-system-1")));
+        }
+    }
+
+    public void testIgnoreUnavailableFalse() {
+        String[] names = new String[] { "missing", "logs*" };
+        IndicesOptions indicesOptions = IndicesOptions.STRICT_EXPAND_OPEN;
+        List<ResolvedIndex> indices = new ArrayList<>();
+        List<ResolvedAlias> aliases = new ArrayList<>();
+        List<ResolvedDataStream> dataStreams = new ArrayList<>();
+        IndexNotFoundException infe = expectThrows(
+            IndexNotFoundException.class,
+            () -> TransportAction.resolveIndices(names, indicesOptions, clusterState, resolver, indices, aliases, dataStreams)
+        );
+        assertThat(infe.getMessage(), containsString("no such index [missing]"));
+    }
+
+    public void testAllowNoIndices() {
```

Comment:
Nit: Based on the name of the previous test method, should this be
```suggestion
    public void testAllowNoIndicesFalse() {
```
?